### DISCUSSION TITLE
Threading

### DISCUSSION THREAD

**Discussion by BluePython339** [Vector35/binaryninja-api#2191] [Q&A] [answered]
When using multi threading inside the GUI of  Binary ninja, it does not seem to run my functions, as a test I ran the following code inside a plugin:

```
def temper(rando):
	print("hello")
	print(rando)
	return



if __name__ == "__main__":
	threads = []
	for i in range(4):
		a = Thread(target=temper, args=(1,))
		threads.append(a)
	for i in threads:
		i.start()
	for i in threads:
		i.join()
```
it doesnt print anything, however if i run it using a normal python shell it does work properly. Is the gui blokking multithreading?

**Comment by negasora** [comment, marked as answer]
The functions run, but we don't connect to stdin/stdout/stderr of threads spawned from the console - thus no output.
You can verify this via

```python
from threading import Thread
Thread(target=log_alert, args=('test',)).start()
```
